pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -387,9 +387,9 @@ func (config *V1RegistriesConf) ConvertToV2() (*V2RegistriesConf, error) {
 // anchoredDomainRegexp is an internal implementation detail of postProcess, defining the valid values of elements of UnqualifiedSearchRegistries.
 var anchoredDomainRegexp = regexp.Delayed("^" + reference.DomainRegexp.String() + "$")
 
-// postProcess checks the consistency of all the configuration, looks for conflicts,
+// PostProcessRegistries checks the consistency of all the configuration, looks for conflicts,
 // and normalizes the configuration (e.g., sets the Prefix to Location if not set).
-func (config *V2RegistriesConf) postProcessRegistries() error {
+func (config *V2RegistriesConf) PostProcessRegistries() error {
```

Comment:
I don’t think it makes sense to expose the output of this utility as a stable API; the internal parsing step might very well create a completely different in-memory format (plausibly some kind of trie to speed up lookups).

We would also end up with two different `V2RegistriesConf` “data types”, with different semantics and validation rules, that are represented as the same Go type.

Reply:
Admittedly I wasn't sold on exposing `V2RegistriesConf.PostProcessRegistries()`.

What if instead we had a specific `sysregistriesv2.PostProcessV2RegistriesConf(config)` or `sysregistriesv2.CleanConfig(config)` or something of that nature?  With that the `V2RegistriesConf` object stays as is, existing users of the API will not be impacted unless using the two new API methods (`FindRegistiesWithConfig` and `CleanConfig`).  

We can be clear in the docs these methods and maintaining a V2RegistriesConf externally is 'discouraged' (for the reasons mentioned).  That would buy us time until the when the `v3` re-write happens.